core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -102,6 +106,36 @@ private[spark] case class SSLOptions(
     }
   }
 
+  /*
+   * The supportedAlgorithms set is a subset of the enabledAlgorithms that
+   * are supported by the current Java security provider for this protocol.
+   */
+  private val supportedAlgorithms: Set[String] = {
+    var context: Option[SSLContext] = Some(SSLContext.getDefault)
```

Comment:
I think that's more straightforward yes. When do you get the NPE? is it more straightforward to just check if protocol is set rather than turn it into null and then wait for something to barf? or is that not the logic.

So the change here is mostly to remove unsupported algos from the list of enabled ones? that makes sense